docs/website/docs/examples/chess_production/code/chess-snippets.py

```diff
@@ -0,0 +1,183 @@
+
+        # this resource takes data from players and returns profiles
+        # it uses `defer` decorator to enable parallel run in thread pool.
+        # defer requires return at the end so we convert yield into return (we return one item anyway)
+        # you can still have yielding transformers, look for the test named `test_evolve_schema`
+        @dlt.transformer(data_from=players, write_disposition="replace")
+        @dlt.defer
+        def players_profiles(username: Any) -> TDataItems:
+            print(
+                f"getting {username} profile via thread {threading.current_thread().name}"
+            )
+            sleep(1)  # add some latency to show parallel runs
+            return _get_data_with_retry(f"player/{username}")
+
+        # this resource takes data from players and returns games for the last month
+        # if not specified otherwise
+        @dlt.transformer(data_from=players, write_disposition="append")
+        def players_games(username: Any) -> Iterator[TDataItems]:
+            # https://api.chess.com/pub/player/{username}/games/{YYYY}/{MM}
+            path = f"player/{username}/games/{year:04d}/{month:02d}"
+            yield _get_data_with_retry(path)["games"]
+
+        return players(), players_profiles, players_games
+
+    # @@@DLT_SNIPPET_END markdown_source
+
+    # @@@DLT_SNIPPET_START markdown_retry_cm
+    from tenacity import (
+        Retrying,
+        retry_if_exception,
+        stop_after_attempt,
+        wait_exponential,
+    )
+
+    from dlt.common import logger
+    from dlt.common.runtime.slack import send_slack_message
+    from dlt.pipeline.helpers import retry_load
+
+    MAX_PLAYERS = 5
+
+    def load_data_with_retry(pipeline, data):
+        try:
+            for attempt in Retrying(
+                stop=stop_after_attempt(5),
+                wait=wait_exponential(multiplier=1.5, min=4, max=10),
+                retry=retry_if_exception(retry_load(())),
+                reraise=True,
+            ):
+                with attempt:
+                    logger.info(
+                        f"Running the pipeline, attempt={attempt.retry_state.attempt_number}"
+                    )
+                    load_info = pipeline.run(data)
+                    logger.info(str(load_info))
+                    # raise on failed jobs
+                    load_info.raise_on_failed_jobs()
+                    # send notification
+                    send_slack_message(
+                        pipeline.runtime_config.slack_incoming_hook,
+                        "Data was successfully loaded!"
+                    )
+        except Exception:
+            # we get here after all the failed retries
+            # send notification
+            send_slack_message(
+                pipeline.runtime_config.slack_incoming_hook,
+                "Something went wrong!"
+            )
+            raise
+        finally:
```

Comment:
put it in `else`, not `finally` or just do it outside of exception handler. you make sure that we raise in `except` so it will be fine